posthog/async_migrations/migrations/0006_persons_and_groups_on_events_backfill.py

```diff
@@ -0,0 +1,259 @@
+                    group1_created_at=dictGetDateTime('groups_dict', 'created_at', tuple(team_id, 1, $group_1)),
+                    group2_created_at=dictGetDateTime('groups_dict', 'created_at', tuple(team_id, 2, $group_2)),
+                    group3_created_at=dictGetDateTime('groups_dict', 'created_at', tuple(team_id, 3, $group_3)),
+                    group4_created_at=dictGetDateTime('groups_dict', 'created_at', tuple(team_id, 4, $group_4))
+                WHERE person_id = ''
+            """,
+            settings={"mutations_sync": 1, "max_execution_time": 0},
+        )
+
+    @cached_property
+    def operations(self):
+        return [
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    CREATE TABLE {TEMPORARY_PERSONS_TABLE_NAME}
+                    (
+                        `id` UUID,
+                        `created_at` DateTime,
+                        `team_id` Int64,
+                        `properties` String,
+                        `is_deleted` Int8 DEFAULT 0,
+                        `version` UInt64
+                    )
+                    ENGINE = ReplacingMergeTree(version)
+                    ORDER BY (team_id, id)
+                    SETTINGS index_granularity = 128
+                """,
+                rollback=f"DROP TABLE IF EXISTS {TEMPORARY_PERSONS_TABLE_NAME}",
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    CREATE TABLE {TEMPORARY_PDI2_TABLE_NAME}
+                    (
+                        `team_id` Int64,
+                        `distinct_id` String,
+                        `person_id` UUID,
+                        `is_deleted` Int8,
+                        `version` Int64 DEFAULT 1
+                    )
+                    ENGINE = ReplacingMergeTree(version)
+                    ORDER BY (team_id, distinct_id)
+                    SETTINGS index_granularity = 128
+                """,
+                rollback=f"DROP TABLE IF EXISTS {TEMPORARY_PDI2_TABLE_NAME}",
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    CREATE TABLE {TEMPORARY_GROUPS_TABLE_NAME}
+                    (
+                        `group_type_index` UInt8,
+                        `group_key` String,
+                        `created_at` DateTime64(3),
+                        `team_id` Int64,
+                        `group_properties` String,
+                        `_timestamp` DateTime
+                    )
+                    ENGINE = ReplacingMergeTree(_timestamp)
+                    ORDER BY (team_id, group_type_index, group_key)
+                    SETTINGS index_granularity = 128
+                """,
+                rollback=f"DROP TABLE IF EXISTS {TEMPORARY_GROUPS_TABLE_NAME}",
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
```

Comment:
This is missing settings - it will time out.

At minimum, we should set `max_execution_time` to 0, probably also `optimize_on_insert` and `max_insert_threads`. Original issue has some good default values.